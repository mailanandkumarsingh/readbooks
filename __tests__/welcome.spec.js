import React from 'react';
import Welcome from '../src/views/welcome';
import Chart from '../src/components/chart';
import renderer from 'react-test-renderer';

describe('Welcome', () => {
  it('renders Welcome coponent correctly', () => {
    const tree = renderer.create(
      <Welcome name="World" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty BarChart component correctly', () => {
    const barChart = renderer.create(
      <Chart data="[]" /> 
      //Passing an array of 40 elements similar to what is being 
      // returned by the 5 days weather app. This is a snapshot capture
    ).toJSON();
    expect(barChart).toMatchSnapshot();
  });

});
