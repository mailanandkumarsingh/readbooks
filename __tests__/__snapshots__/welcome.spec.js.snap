// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders Welcome coponent correctly 1`] = `
<h1>
  Hello to you, 
  World
</h1>
`;

exports[`Welcome renders empty BarChart component correctly 1`] = `
<div>
  <div
    className="rd3-basic-chart"
  >
    <svg
      className="rd3-chart"
      height={250}
      viewBox={undefined}
      width={350}
    >
      <g
        className="rd3-barchart"
        transform="translate(45,10)"
      >
        <g
          className="rd3-barchart-yaxis"
          transform="translate(0, 0)"
        >
          <g />
          <path
            className="domain"
            d="M-6,0H0V200H-6"
            fill="none"
            stroke="#000"
            strokeWidth="1"
            style={
              Object {
                "shapeRendering": "crispEdges",
              }
            }
          />
          <text
            strokeWidth="0.01"
            textAnchor="middle"
            transform="rotate(270)"
            x={-100}
            y={-35}
          >
            Temperature
          </text>
        </g>
        <g
          className="rd3-barchart-xaxis"
          transform="translate(0 ,200)"
        >
          <g>
            <g
              className="tick"
              transform="translate(142.5,0)"
            >
              <line
                style={
                  Object {
                    "opacity": "1",
                    "shapeRendering": "crispEdges",
                    "stroke": "#000",
                  }
                }
                x2={undefined}
                y2={6}
              />
              <text
                dy=".71em"
                strokeWidth="0.01"
                style={
                  Object {
                    "fill": undefined,
                    "stroke": undefined,
                  }
                }
                textAnchor="middle"
                x={undefined}
                y={9}
              >
                Day 1
              </text>
            </g>
          </g>
          <path
            className="domain"
            d="M0,6V0H285V6"
            fill="none"
            stroke="none"
            strokeWidth="1"
            style={
              Object {
                "shapeRendering": "crispEdges",
              }
            }
          />
          <text />
        </g>
        <g>
          <rect
            className="rd3-barchart-bar"
            fill="#3182bd"
            handleMouseLeave={[Function]}
            handleMouseOver={[Function]}
            height={NaN}
            hoverAnimation={true}
            offset={0}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            width={171}
            x={57}
            y={NaN}
          />
        </g>
      </g>
    </svg>
  </div>
</div>
`;
